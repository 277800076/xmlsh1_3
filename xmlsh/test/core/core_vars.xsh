# core_vars.xsh
# test variables


# clear just in case they were exported
unset A AA

echo NO $A variable
echo NO $AA variable
echo NO ${A} variable
echo NO ${AA} variable

A=foo
AA=bar

echo $A variable
echo $AA variable
echo ${AA} variable

set A $A "$A"

if [ $3 != foo -o $3 != "foo" ] ; then
	echo $3 should be foo
fi


# Test $()

A=$(echo '<foo/>' | xcat)
echo $A 


# Test local variable setting for simple commands
unset A
A=B eval 'echo A is $A'
[ -z "$A" ] || echo A should not be set


exit 0