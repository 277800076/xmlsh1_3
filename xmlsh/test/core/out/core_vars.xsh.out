NO variable
NO variable
NO variable
NO variable
foo variable
bar variable
bar variable
<foo/>
A is B
